builtInFunctions/esdtNFTCreate_test.go

```diff
@@ -444,80 +450,206 @@ func TestEsdtNFTCreate_ProcessBuiltinFunctionWithExecByCallerCrossChainToken(t *
 
 	// Nonce was not saved in account
 	nonceKey := getNonceKey([]byte(token))
-	latestNonceBytes, _, err := mock.NewUserAccount(userAddr).AccountDataHandler().RetrieveValue(nonceKey)
+	latestNonceBytes, _, err := whiteListedAcc.AccountDataHandler().RetrieveValue(nonceKey)
 	require.Nil(t, err)
 	latestNonce := big.NewInt(0).SetBytes(latestNonceBytes).Uint64()
 	require.Zero(t, latestNonce)
 
-	tokenMetaData := &esdt.MetaData{
-		Nonce:      nonce.Uint64(),
-		Name:       []byte(name),
-		Creator:    originalCreator,
-		Royalties:  uint32(royalties),
-		Hash:       hash,
-		URIs:       uris,
-		Attributes: attributes,
+	// check metadata from vm output
+	esdtDataBytes := vmOutput.Logs[0].Topics[3]
+	var esdtDataFromLog esdt.ESDigitalToken
+	err = nftCreate.marshaller.Unmarshal(&esdtDataFromLog, esdtDataBytes)
+	require.Nil(t, err)
+	expectedMetaEsdt := &esdt.ESDigitalToken{
+		Type:  uint32(tokenType),
+		Value: quantity,
+		TokenMetaData: &esdt.MetaData{
+			Nonce:      nonce.Uint64(),
+			Name:       []byte(name),
+			Creator:    originalCreator,
+			Royalties:  uint32(royalties),
+			Hash:       hash,
+			URIs:       uris,
+			Attributes: attributes,
+		},
 	}
-
-	tokenKey := []byte(baseESDTKeyPrefix + token)
-	tokenKey = append(tokenKey, nonce.Bytes()...)
-
-	metaData, _ := esdtDtaStorage.getESDTMetaDataFromSystemAccount(tokenKey, defaultQueryOptions())
-	assert.Equal(t, tokenMetaData, metaData)
+	require.Equal(t, expectedMetaEsdt, &esdtDataFromLog)
```

Comment:
I think this check that you deleted was important?
`metaData, _ := esdtDtaStorage.getESDTMetaDataFromSystemAccount(tokenKey, defaultQueryOptions())`

Reply:
I remember I had a pointer issue with whitelistedAcc mock because is an execute on dest call. I found a solution and fixed it.